plutus-core/src/Language/PlutusCore/Constant/Typed.hs

```diff
@@ -155,6 +162,15 @@ proceeds as follows:
 Note how we use the 'Opaque' wrapper in order to unlift a PLC term as an opaque Haskell value
 using 'readKnown' and then lift the term back using 'makeKnown' without ever inspecting the term.
 
+An opaque PLC @term@ whose type is a PLC type variable `a_0` has the following type on the Haskell
+side:
+
+    Opaque term (TyVarRep ('TyNameRep "a" 0))
```

Comment:
> Should this be `... ('TyNameRep @k "a" 0)...` where `a_0 :: k`?

No, `Opaque term rep` constrains `rep` to be of kind `*`, so we don't need to specify it. And I specifically wanted to avoid talking about the intrinsic kindedness there just yet.